### DISCUSSION TITLE
Package name mismatch found while reading

### DISCUSSION THREAD

**Discussion by tjx666** [pnpm/pnpm#4359] [Q&A] [answered]
@zkochan 
<img width="1083" alt="image" src="https://user-images.githubusercontent.com/41773861/154796236-cdb843ee-202a-43a4-bc20-f2c492d45ebf.png">

Version:
pnpm: 6.31.0
node: 16.13.2
os: MacOS12.2.1

Key info:

> Expected package: filelist@1.0.2. Actual package in the store by the given integrity: FileList@1.0.2.

I have no idea about how to fix it.

**Comment by zkochan** [comment]
Maybe we should compare the package names ignoring case

https://github.com/pnpm/pnpm/blob/cf478bb8b31c23203f0a0795bbae232245a62dd4/packages/package-requester/src/packageRequester.ts#L449

**Comment by zkochan** [comment, marked as answer]
This is fixed in pnpm v6.32

**Reply by gxmzjxk** [reply]
![image](https://user-images.githubusercontent.com/16485400/158768919-7a9c1ed5-2bbe-4830-a25e-5de6125a9f29.png)

**Reply by gxmzjxk** [reply]
I still meet this question in 6.32.3

**Reply by zkochan** [reply]
Looks like your lockfile is broken. And looks like you use `--frozen-lockfile`? Try `pnpm install --no-frozen-lockfile`.

If it doesn't help, try to remove `pnpm-lock.yaml` and `node_modules` and then run `pnpm install`.

**Comment by mdarche** [comment]
Ran into a similar issue when updating a private package dependency. I fixed by updating the store with `pnpm store prune` and then deleting `node_modules` and `pnpm-lock.yaml` like @zkochan suggested. After a fresh install, all mismatch errors went away.

**Reply by zhongjie-chen** [reply]
good
